geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/PRClientServerTestBase.java

```diff
@@ -368,6 +377,36 @@ private static void createNoSingleHopCacheClient(String host, Integer port1, Int
     assertNotNull(region);
   }
 
+  private static void createNoSingleHopCacheClient(String host,
+      Integer port1, Integer port2, Integer port3, int connectTimeout) {
+    CacheServerTestUtil.disableShufflingOfEndpoints();
+
+    Pool p;
+    try {
+      PoolFactory factory = PoolManager.createFactory().addServer(host, port1)
+          .addServer(host, port2).addServer(host, port3).setPingInterval(2000)
+          .setSubscriptionEnabled(true).setSubscriptionRedundancy(-1).setReadTimeout(2000)
+          .setSocketBufferSize(1000).setRetryAttempts(0)
+          .setSocketConnectTimeout(connectTimeout)
+          .setPRSingleHopEnabled(false);
+      if (connectTimeout > 0) {
+        factory.setSocketConnectTimeout(connectTimeout);
+      }
```

Comment:
setConnectionTimeout is already being done right above this, so this check isn't necessary.